### PR TITLE
enable tfstate plugin

### LINKED ISSUE #31: enable tfstate plugin?
want refer tfstate value as well as ecspresso.
can i implement it?

or may I implement it?

### PR DESCRIPTION
**Pull request by gotoeveryone**
close #31

- can be enable tfstate plugin as well as way of ecspresso
- replace to tfstate's values using a `tfstate` and `tfstatef` functions
  - replace tfstate ref to `ecschedule::tfstate*` value when parse yaml
  - lookup tfstate for `ecschedule::tfstate*` value when after parsed

### CODE REVIEW

**Comment by gotoeveryone** [comment]
Do you have any other good ideas?

File: rule.go [hunk 1]
```diff
@@ -296,6 +297,25 @@ func (r *Rule) validateEnv() error {
...
+	m := tfstateReg.FindAllSubmatch(bs, -1)
+	if len(m) > 0 {
+		if len(m) == 1 {
+			return fmt.Errorf("tfstate reference %s is not defined", string(m[0][1]))
+		}
+		var envs []string
+		for _, v := range m {
+			envs = append(envs, string(v[1]))
+		}
+		return fmt.Errorf("tfstate reference %s are not defined", strings.Join(envs, " and "))
```
Comment (gotoeveryone) [inline rule.go:314]: @Songmu 
Do you have any other good ideas?

File: template.go [hunk 2]
```diff
@@ -32,6 +36,12 @@ func init() {
 			}
 			return fmt.Sprintf("ecschedule::<%s>", key)
 		},
+		"tfstate": func(key string) string {
+			return fmt.Sprintf("ecschedule::tfstate::<`%s`>", key)
+		},
+		"tfstatef": func(key string, args ...string) string {
+			return fmt.Sprintf("ecschedule::tfstatef::<`%s` `%s`>", key, strings.Join(args, "` `"))
```
Comment (gotoeveryone) [inline template.go:43]: @Songmu 
Since we receive any arguments, we use split and join to maintain variable names...  
Do you have any other good ideas?

**Comment by Songmu** [comment]
Thanks. tfstate plugin is a good idea. Thanks for promoting the implementation as well.
May I proceed with reviewing the current implementation and suggestions for questions? If so, please assign me as a reviewer. @gotoeveryone

**Comment by gotoeveryone** [comment]
@Songmu 
Sorry, I can't assign reviewers...
image

**Comment by gotoeveryone** [comment]
What are the detailed steps to PR merge?
Add the required Mention....

**Comment by Songmu** [comment]
Oops, sorry... In any case, I will take a look at this pull request this weekend.

**Comment by Songmu** [comment]
Thank you. I will merge it once and release. I would appreciate your feedback on its use.

**Pull request merged**